src/libs/utils.js

```diff
@@ -63,3 +63,22 @@ export const cond = function(...args) {
 
   return maybeCall(match ? match[1] : defaultValue)
 }
+
+/**
+ * Memoizes the given function, but expires after the specified duration (ms).
+ * Uses the first argument of the function as the cache key.
```

Comment:
> Uses the first argument of the function as the cache key.

Doesn't this need to be a combination of all arguments?

Reply:
Still strikes me as a bug. Seems like we can simply make the cache key the entire array of arguments.

Reply:
I see benefit both ways (using all vs. matching an existing signature from elsewhere). Since an anonymous function would break things, is it worth asserting against that case? Is that an argument for only using the first arg as the key?

Reply:
I just feel like this is shockingly easy to break:

https://repl.it/repls/RigidWordyFormula

Reply:
I find the example to be a compelling argument.